project/recursive_sorting.py

```diff
@@ -36,20 +42,45 @@ def merge_in_place(arr, start, mid, end):
 
     return arr
 
-def merge_sort_in_place(arr, l, r): 
+
+def merge_sort_in_place(arr, l, r):
     # TO-DO
 
     return arr
 
 
 # TO-DO: implement the Quick Sort function below USING RECURSION
-def quick_sort( arr, low, high ):
-
+def quick_sort(arr, low, high):
+    if(low >= high):
+        return arr
+    # Use high as pivot
+    index_pivot = high
+    index_low = low
+    # Iterate over test cases, from case previous to high to lowest.
+    value_pivot = arr[index_pivot]
+    while(index_low < index_pivot):
+        # Always test the value directly before pivot
+        index_test = index_pivot-1
+        value_test = arr[index_test]
+        # If test is greater than pivot, swap them. Then Advance pivot index.
+        if(value_test > value_pivot):
+            arr[index_pivot], arr[index_test] = value_test, value_pivot
+            index_pivot = index_test
+        # Otherwise, swap test with low value. Then advance low index.
+        else:
+            value_low = arr[index_low]
+            arr[index_low], arr[index_test] = value_test, value_low
```

Comment:
this also works

arr[index_low], arr[index_test] = arr[index_test], arr[index_low]